pkg/agent/watch.go

```diff
@@ -317,6 +317,21 @@ func makeVMMemMetrics(vm *vmapi.VirtualMachine) []vmMetric {
 	return metrics
 }
 
+func makeVMRestartCountMetrics(vm *vmapi.VirtualMachine) []vmMetric {
```

Comment:
wdyt about just returning a single `vmMetric` here? we'd have to wrap it in a `[]vmMetric` inside of `updateVMMetrics` but otherwise would be simpler imo

(as I was reading through the other functions, I was confused why we were iterating over a single restart count, then realized this returned a slice)

Reply:
I returned a slice to keep the code similar to other metrics in all places and avoid handling whether the metric exist (since `.status.restartCount` may be nil). But I agree this is a trade-off. wdyt?

Reply:
Hm, yeah fair enough — I forgot `.status.restartCount` can be nil (_maybe_ we'd want to use a value of zero when it's nil? wdyt?). Either way ok by me

Reply:
One way around this would be call it `makeVMRestartMetrics`, without the `Count` word. This way it suggests there can be more metrics in the future e.g. `TotalRestartTimeSeconds`

Also can add smth like 
```
// Only one metric for now
```